implementations/micrometer-registry-influx/src/main/java/io/micrometer/influx/InfluxMeterRegistry.java

```diff
@@ -232,6 +249,84 @@ private String influxLineProtocol(Meter.Id id, String metricType, Stream<Field>
                 + " " + clock.wallTime();
     }
 
+    private void recognizeInfluxDBVersion() {
+
+        if (influxDBVersion != null) {
+            return;
+        }
+
+        String uri = config.uri() + "/ping";
+        try {
+            httpClient
+                    .head(uri)
+                    .send()
+                    .onSuccess(response -> {
+                        logger.debug("InfluxDB successfully pinged: '{}'", response.headers());
+                        influxDBVersion = response.headers().entrySet()
+                                .stream()
+                                .filter(entry -> "X-Influxdb-Version".equalsIgnoreCase(entry.getKey()))
+                                .map(entry -> {
+                                    boolean v1 = entry.getValue().stream().anyMatch(value -> value.startsWith("1."));
+                                    return v1 ? InfluxDBVersion.V1 : InfluxDBVersion.V2;
+                                })
+                                .findFirst()
+                                // There is no header "X-Influxdb-Version: 1.X" => use v2 API
+                                .orElse(InfluxDBVersion.V2);
+                        logger.debug("InfluxDB version configured to: '{}'", influxDBVersion);
+                    })
+                    .onError(response -> {
+                        logger.error("unable to ping InfluxDB: '{}'. Use v2 API.: {}", uri, response.body());
+                        // Unable ping http://server/ping => use v2 API
+                        influxDBVersion = InfluxDBVersion.V2;
+                    });
+        } catch (Throwable e) {
+            logger.error("unable to ping InfluxDB: '{}', Use v1 API.", uri, e);
+            influxDBVersion = InfluxDBVersion.V1;
+        }
+
+        if (influxDBVersion.equals(InfluxDBVersion.V2)) {
+            org = config.org();
+            if (StringUtils.isBlank(org)) {
+                retrieveOrg();
+            }
+
+            checkAll(config,
+                    c -> StepRegistryConfig.validate(c),
+                    checkRequired("token", InfluxConfig::token).andThen(Validated::nonBlank),
+                    checkRequired("org", (Function<InfluxConfig, String>) it -> org)
+                            .andThen(Validated::nonBlank)
+            ).orThrow();
+        }
+    }
+
+    private void retrieveOrg() {
+        String uri = config.uri() + "/api/v2/orgs";
```

Comment:
As far as I can tell, only an admin or all access token can read the orgs. A read/write token (which is what I would expect to be provided for applications to write metrics) will not get a response with any orgs, even the one it belongs to. Here is what I get back with a read/write token:

```
{
    "links": {
        "self": "/api/v2/orgs"
    },
    "orgs": []
}
```

While it would be nice for users to have one less thing to configure, I am not sure it is worth having this attempt to retrieve the org. It also adds the JSON parser dependency. Users will have to configure the bucket and token anyways, so it isn't like we can provide something that works with defaults alone.

What do you think about removing the retrieve org logic?

Reply:
> What do you think about removing the retrieve org logic?

I agree. 

It is same case as a creating a `bucket` => the support for retrieve `org` is removed.